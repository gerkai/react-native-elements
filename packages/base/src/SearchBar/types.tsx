import {
  PressableProps,
  ViewStyle,
  StyleProp,
  TextStyle,
  ActivityIndicatorProps,
} from 'react-native';
import { Theme } from '../helpers';
import { IconNode } from '../Icon';
import { InputProps } from '../Input';

export type SearchBarAndroidProps = SearchBarBaseProps & {
  platform?: 'android';
  cancelIcon?: IconNode;
  onCancel?: () => any;
};

export type SearchBarIosProps = SearchBarBaseProps & {
  platform?: 'ios';
  cancelButtonProps?: Partial<PressableProps> & {
    buttonStyle?: StyleProp<ViewStyle>;
    buttonTextStyle?: StyleProp<TextStyle>;
    color?: string;
    buttonDisabledStyle?: StyleProp<ViewStyle>;
    buttonDisabledTextStyle?: StyleProp<ViewStyle>;
  };
  cancelButtonTitle?: string;
  showCancel?: boolean;
  searchIcon?: IconNode;
  clearIcon?: IconNode;
  /**
   * @type Callback Function on cancel icon press
   */
  onCancel?: () => any;
};

export type SearchBarDefaultProps = SearchBarBaseProps & {
  platform?: 'default';
  lightTheme?: boolean;
  round?: boolean;
};

export interface SearchBarBaseProps extends InputProps {
  containerStyle?: StyleProp<ViewStyle>;
  inputContainerStyle?: StyleProp<ViewStyle>;
  clearIcon?: IconNode;
  searchIcon?: IconNode;
  inputStyle?: StyleProp<TextStyle>;
  loadingProps?: ActivityIndicatorProps;
  showLoading?: boolean;
  leftIconContainerStyle?: StyleProp<ViewStyle>;
  rightIconContainerStyle?: StyleProp<ViewStyle>;
  /**
   * @type Callback Function on clear icon press
   */
  onClear?(): void;
  theme?: Theme;
}

export type SearchBarProps =
  | SearchBarDefaultProps
  | SearchBarAndroidProps
  | SearchBarIosProps;
