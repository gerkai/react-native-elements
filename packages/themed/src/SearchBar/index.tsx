import { withTheme } from '../config';
import {
  SearchBar,
  SearchBarBaseProps,
} from '@react-native-elements/base/dist/SearchBar/SearchBar';
import { SearchBarAndroidProps } from '@react-native-elements/base/dist/SearchBar/SearchBar-android';
import { SearchBarIosProps } from '@react-native-elements/base/dist/SearchBar/SearchBar-ios';
import { SearchBarDefaultProps } from '@react-native-elements/base/dist/SearchBar/SearchBar-default';

export { SearchBar };
export type {
  SearchBarBaseProps as SearchBarProps,
  SearchBarAndroidProps,
  SearchBarDefaultProps,
  SearchBarIosProps,
};
export default withTheme<SearchBarBaseProps>(SearchBar, 'SearchBar');
