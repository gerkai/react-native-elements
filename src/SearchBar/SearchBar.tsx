import React from 'react';
import { SearchBarIOS, SearchBarIosProps } from './SearchBar-ios';
import { SearchBarAndroid, SearchBarAndroidProps } from './SearchBar-android';
import { SearchBarDefault, SearchBarDefaultProps } from './SearchBar-default';
import {
  ActivityIndicatorProps,
  StyleProp,
  TextStyle,
  ViewStyle,
  TextInput,
} from 'react-native';
import { IconNode } from '../Icon';
import { ThemeProps } from '../config';

const SEARCHBAR_COMPONENTS = {
  ios: SearchBarIOS,
  android: SearchBarAndroid,
  default: SearchBarDefault,
};

export type SearchBarBaseProps = React.ComponentPropsWithRef<
  typeof TextInput
> & {
  platform?: 'default' | 'ios' | 'android';
  containerStyle?: StyleProp<ViewStyle>;
  inputContainerStyle?: StyleProp<ViewStyle>;
  clearIcon?: IconNode;
  searchIcon?: IconNode;
  inputStyle?: StyleProp<TextStyle>;
  loadingProps?: ActivityIndicatorProps;
  showLoading?: boolean;
  leftIconContainerStyle?: StyleProp<ViewStyle>;
  rightIconContainerStyle?: StyleProp<ViewStyle>;
  onClear?(): void;
  onFocus?(): void;
  onBlur?(): void;
  onChangeText?(text: string): void;
  onCancel?(): void;
};

export type SearchBarProps =
  | SearchBarBaseProps
  | SearchBarDefaultProps
  | SearchBarAndroidProps
  | SearchBarIosProps;

export class SearchBar extends React.Component<
  SearchBarProps & Partial<ThemeProps<SearchBarProps>>
> {
  searchbar!: SearchBarIOS;
  static defaultProps = {
    platform: 'default' as const,
  };

  focus = () => {
    this.searchbar.focus();
  };

  blur = () => {
    this.searchbar.blur();
  };

  clear = () => {
    this.searchbar.clear();
  };

  cancel = () => {
    this.searchbar.cancel && this.searchbar.cancel();
  };

  render() {
    const Component: typeof React.Component =
      SEARCHBAR_COMPONENTS[this.props.platform] || SearchBarDefault;

    return (
      <Component
        ref={(ref: SearchBarIOS) => {
          this.searchbar = ref;
        }}
        {...this.props}
      />
    );
  }
}
