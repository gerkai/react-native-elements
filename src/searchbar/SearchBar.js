import PropTypes from 'prop-types';
import React, { Component } from 'react';
import IOSSearchBar from './SearchBar-ios';
import AndroidSearchBar from './SearchBar-android';
import DefaultSearchBar from './SearchBar-default';

const SEARCHBAR_COMPONENTS = {
  ios: IOSSearchBar,
  android: AndroidSearchBar,
  default: DefaultSearchBar,
};

class SearchBar extends Component {

  focus = () => {
    this.searchbar.focus();
  };

  blur = () => {
    this.searchbar.blur();
  };

  clear = () => {
    this.searchbar.clear();
  };

  cancel = () => {
    this.searchbar.cancel && this.searchbar.cancel();
  };


  render() {
    const SearchBar = SEARCHBAR_COMPONENTS[this.props.platform] || DefaultSearchBar;
    return <SearchBar ref={searchbar => this.searchbar = searchbar} {...this.props} />;
  }
}

SearchBar.propTypes = {
  platform: PropTypes.oneOf(['default', 'ios', 'android']),
};

SearchBar.defaultProps = {
  platform: 'default',
};


export default SearchBar;