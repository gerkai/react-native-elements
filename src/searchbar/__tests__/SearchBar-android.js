import React from 'react';
import { shallow } from 'enzyme';
import { View } from 'react-native';
import toJson from 'enzyme-to-json';
import SearchBar from '../SearchBar-android';

describe('Android SearchBar component', () => {
  it('should render without issues', () => {
    const component = shallow(<SearchBar />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with loading', () => {
    const component = shallow(
      <SearchBar
        showLoading
        loadingProps={{
          style: { flex: 1 },
        }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with a custom search icon component', () => {
    const component = shallow(<SearchBar searchIcon={<View />} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with a custom search icon', () => {
    const component = shallow(<SearchBar searchIcon={{ size: 50 }} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without search icon', () => {
    const component = shallow(<SearchBar searchIcon={false} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with a custom clear icon', () => {
    const component = shallow(<SearchBar clearIcon={{ color: 'black' }} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with a custom clear icon component', () => {
    const component = shallow(<SearchBar clearIcon={<View />} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without clear icon', () => {
    const component = shallow(<SearchBar clearIcon={false} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with a custom cancel icon', () => {
    const component = shallow(<SearchBar cancelIcon={{ color: 'black' }} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with a custom cancel icon component', () => {
    const component = shallow(<SearchBar cancelIcon={<View />} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without cancel icon', () => {
    const component = shallow(<SearchBar cancelIcon={false} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onFocus when input is focused', () => {
    const onFocusMock = jest.fn();
    const component = shallow(<SearchBar onFocus={onFocusMock} />);
    component.find('Input').simulate('focus');
    expect(onFocusMock).toBeCalled();
  });

  it('should call onBlur when input is blured', () => {
    const onBlurMock = jest.fn();
    const component = shallow(<SearchBar onFocus={onBlurMock} />);
    component.find('Input').simulate('focus');
    component.find('Input').simulate('blur');
    expect(onBlurMock).toBeCalled();
  });

  it('should call onChangeText when input is changed', () => {
    const onChangeMock = jest.fn();
    const component = shallow(<SearchBar onChangeText={onChangeMock} />);
    component.find('Input').simulate('changeText', 'test');
    expect(onChangeMock).toBeCalled();
  });
  it('should render with a custom methods', () => {
    const onChangeTextMock = jest.fn();
    const onClearMock = jest.fn();
    const onCancelMock = jest.fn();
    const component = shallow(
      <SearchBar
        onChangeText={onChangeTextMock}
        onClear={onClearMock}
        onCancel={onCancelMock}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
