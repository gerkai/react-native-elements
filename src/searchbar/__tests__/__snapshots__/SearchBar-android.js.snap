// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android SearchBar component should render with a custom cancel icon 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with a custom cancel icon component 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with a custom clear icon 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with a custom clear icon component 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with a custom methods 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[MockFunction]}
    onChangeText={[Function]}
    onClear={[MockFunction]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with a custom search icon 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={50}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with a custom search icon component 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={<Component />}
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render with loading 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
          style={
            Array [
              Object {
                "marginRight": 5,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </Component>
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render without cancel icon 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render without clear icon 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render without issues 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={
      <Icon
        color="rgba(0, 0, 0, 0.54)"
        disabled={false}
        name="magnify"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={25}
        type="material-community"
        underlayColor="white"
      />
    }
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;

exports[`Android SearchBar component should render without search icon 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Input
    inputContainerStyle={
      Array [
        Object {
          "borderBottomWidth": 0,
          "width": 750,
        },
        undefined,
      ]
    }
    inputStyle={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 8,
        },
        undefined,
      ]
    }
    leftIcon={null}
    leftIconContainerStyle={
      Array [
        Object {
          "marginLeft": 8,
        },
        undefined,
      ]
    }
    onBlur={[Function]}
    onCancel={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    rightIcon={
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
    }
    rightIconContainerStyle={
      Array [
        Object {
          "marginRight": 8,
        },
        undefined,
      ]
    }
  />
</Component>
`;
